### ISSUE TITLE
What if predict(network) has multiple outputs?

### ISSUE THREAD

**Issue by CandiceUIC** [BorealisAI/advertorch#33]
Hi there, thanks for your awesome work!
I am now facing a problem that my network(predict function) has 2 outputs.

out1, out2 = net(input)
adversary = GradientSignAttack(net,loss_fn,eps,clip_min,clip_max)

GradientSIgnAttack does not work because 2 outputs is confusing. I was thinking about using concatenation (mentioned in another issue), but it might not be helpful. Because my network has 6 constitutional layers, out1 is extracted out after 3rd layer and out2 is the final output. What I need is the adversarial example of out2. If using concatenation, the calculated adversarial example will not be correct, right?

Do you have any idea to deal with this problem?

**Comment by CandiceUIC** [comment]
I solved it by changing a bit of source code. LOL

**Comment by gwding** [comment]
Hi, I think in this case, by our design, the easiest way is to pass in a customized `loss_fn`, that takes both `out1` and `out2` as input. For it example, you can define it like this, if both of them are like logits.
```
ce_loss_fn = nn.CrossEntropyLoss(reduction="sum")

def loss_fn(lst_out, y):
    return ce_loss_fn(lst_out[0], y) + ce_loss_fn(lst_out[1], y)
```

similarly, you can also put a `lst_y` there if your label has structure.

**Comment by gwding** [comment]
Closing this issue for now. Please reopen if needed.
